gson/src/test/java/org/baeldung/gson/deserialization/HashMapDeserializationUnitTest.java

```diff
@@ -0,0 +1,87 @@
+package org.baeldung.gson.deserialization;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.internal.LinkedTreeMap;
+import org.baeldung.gson.entities.Employee;
+import org.baeldung.gson.serialization.HashMapDeserializer;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+
+public class HashMapDeserializationUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(HashMapDeserializationUnitTest.class);
+
+    @Test
+    public void whenUsingHashMapClass_thenShouldReturnMapWithDefaultClasses() {
+
+        String jsonString = "{'employee.name':'Bob','employee.salary':10000, 'employee.active':true, "
+                + "'employee':{'id':10, 'name': 'Bob Willis', 'address':'London'}}";
+
+        Gson gson = new Gson();
+        HashMap map = gson.fromJson(jsonString, HashMap.class);
+
+        logger.info("The converted map: {}", map);
+        Assert.assertEquals(4, map.size());
+        Assert.assertEquals(Double.class, map.get("employee.salary").getClass());
+        Assert.assertEquals(LinkedTreeMap.class, map.get("employee").getClass());
+
+    }
+
+    @Test(expected = JsonSyntaxException.class)
+    public void whenUsingJsonStringWithDuplicateKey_thenShouldThrowJsonSyntaxException() {
+
+        String jsonString = "{'employee.name':'Bob', 'employee.name':'Jenny','employee.salary':10000, "
+                + "'employee.active':true, "
```

Comment:
here's another place where we need only two spaces indentation from previous line